.github/workflows/ci-quick.yml

```diff
@@ -0,0 +1,80 @@
+name: Quick CI
+
+# A short version of the CI, designed to run in seconds (<200s) on every push.
+# Runs lints & basic tests on branches named ci/...
+
+on:
+  push:
+    branches:
+      - ci/*
```

Comment:
Changed to an opt-in approach. Basing this on branch naming as the approach which is easiest for users to understand and follow and which leads to easily readable code.